File: src/main/java/com/github/ofrostdev/api/utils/PacketUtils.java
package com.github.ofrostdev.api.utils;

import net.minecraft.server.v1_8_R3.Packet;
import org.bukkit.Bukkit;
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer;
import org.bukkit.entity.Player;

import java.util.Collection;

public final class PacketUtils {

    private PacketUtils() {}

    public static void sendPacket(Player player, Packet<?> packet) {
        if (player != null && player.isOnline() && packet != null) {
            ((CraftPlayer) player).getHandle().playerConnection.sendPacket(packet);
        }
    }

    public static void sendPacket(Collection<? extends Player> players, Packet<?> packet) {
        if (players == null || packet == null) return;
        for (Player player : players) {
            sendPacket(player, packet);
        }
    }

    public static void sendPacketToAll(Packet<?> packet) {
        sendPacket(Bukkit.getOnlinePlayers(), packet);
    }
}

File: src/main/kotlin/com/github/ofrostdev/api/utils/packets/PacketUtils.kt
package com.github.ofrostdev.api.utils.packets

import net.minecraft.server.v1_8_R3.Packet
import org.bukkit.Bukkit
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer
import org.bukkit.entity.Player

object PacketUtils {
    fun sendPacket(player: Player?, packet: Packet<*>?) {
        if (player != null && player.isOnline && packet != null) {
            (player as CraftPlayer).handle.playerConnection.sendPacket(packet)
        }
    }

    fun sendPacket(players: Collection<Player?>?, packet: Packet<*>?) {
        if (players == null || packet == null) return
        for (player in players) {
            sendPacket(player, packet)
        }
    }

    fun sendPacketToAll(packet: Packet<*>?) {
        sendPacket(Bukkit.getOnlinePlayers(), packet)
    }
}